packages/common/src/hooks/useUrlQuery/useHandleQueryParams.ts

```diff
@@ -0,0 +1,74 @@
+import { useUrlQuery } from './useUrlQuery';
+import { Column, SortController } from '@openmsupply-client/common';
+import { FilterController, PaginationController } from '../useQueryParams';
+
+export const useHandleQueryParams = ({
+  sort,
+  pagination,
+  filterKey,
+}: {
+  sort: SortController<any>;
+  pagination: PaginationController;
+  filterKey: string;
+}) => {
+  const { urlQuery, updateQuery } = useUrlQuery();
+  const { onChangeSortBy, sortBy } = sort;
+  const { onChangePage, page } = pagination;
+
+  const updateSortQuery = (column: Column<any>) => {
+    const currentSort = urlQuery?.['sort'];
+    const sort = column.key as string;
+    if (sort !== currentSort) {
+      updateQuery({ sort, dir: 'asc', page: 1 });
+    } else {
+      const dir = column?.sortBy?.direction === 'asc' ? 'desc' : 'asc';
+      updateQuery({ dir });
+    }
+  };
+
+  const updatePaginationQuery = (page: number) => {
+    // Page is zero-indexed in useQueryParams store, so increase it by one
+    updateQuery({ page: page + 1 });
+  };
+
+  const updateFilterQuery = (key: string, value: string) => {
+    updateQuery({ [key]: value });
+  };
+
+  const onChangeUrlQuery = (
+    columns: Column<any>[],
+    filter: FilterController
+  ) => {
+    // Update Sort
+    if (urlQuery?.['sort'] || urlQuery?.['dir']) {
```

Comment:
I do not like this syntax, but Typescript is requiring it. Do you have any issue with setting `"noPropertyAccessFromIndexSignature": false` in tsconfig.json?
Then it can just be standard dot notation: `if (urlQuery?.sort || urlQuery?.dir)...`

Reply:
I would rather not. The complaint that ts is raising is that the `sort` prop might not exist and given that the shape is

```
interface UrlQueryObject {
  [key: string]: string | number | boolean;
}
```

I agree with ts on this one. I can't see how the `urlQuery` can be undefined though, so you don't need the `?` in which case `urlQuery['sort']` looks fine to me.

An alternative would be to define the expected, but optional, props for the query

```
interface UrlQueryObject
  extends Record<string, string | number | boolean | undefined> {
  sort?: string;
  dir?: 'asc' | 'desc';
  page?: number;
  ...
}
```

which allows `    if (urlQuery.sort || urlQuery.dir) {` while forcing unexpected params into the index format.. `urlQuery['tomato']`